### ISSUE TITLE
Error on https://www.shockwave.com/content/superspavegas/sis/Core.swf

### ISSUE THREAD

**Issue by carolmota18** [ruffle-rs/ruffle#12060]
# Error Info
Error name: Error
Error message: panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: cannot recursively acquire mutex', /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/sys/wasm/../unsupported/locks/mutex.rs:20:9
Error stack:
```
Error: panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: cannot recursively acquire mutex', /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/sys/wasm/../unsupported/locks/mutex.rs:20:9
    at imports.wbg.__wbg_new_d258248ed531ff54 (chrome-extension://donbcfbmhbcapadipfkeojnmajbakjdc/dist/339.js:2403:21)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::h1fb1bf8fc190c8e7 (wasm://wasm/02e582f2:wasm-function[9034]:0x8a07a6)
    at std::panicking::rust_panic_with_hook::h85b7f6628c291e12 (wasm://wasm/02e582f2:wasm-function[5991]:0x7ab2bb)
    at core::panicking::panic_fmt::hf5c4cd929d4aaa9e (wasm://wasm/02e582f2:wasm-function[8879]:0x89bfc5)
    at core::panicking::assert_failed_inner::h2aa5e8e7fe201c6e (wasm://wasm/02e582f2:wasm-function[3683]:0x690ac7)
    at core::panicking::assert_failed::hca8aa44402f73536 (wasm://wasm/02e582f2:wasm-function[7917]:0x85d24c)
    at ruffle_core::loader::Loader::movie_loader::{{closure}}::h04ddde01ea776b6b (wasm://wasm/02e582f2:wasm-function[734]:0x26b54f)
    at <ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h744943c0312e1f77 (wasm://wasm/02e582f2:wasm-function[1057]:0x37ba03)
    at wasm_bindgen_futures::queue::Queue::new::{{closure}}::hcde08edc52289706 (wasm://wasm/02e582f2:wasm-function[3854]:0x6afcb8)
    at <dyn core::ops::function::FnMut<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::hbf671330080d87d4 (wasm://wasm/02e582f2:wasm-function[9692]:0x8bd2c6)
```

# Player Info
Allows script access: false
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "ANGLE (Intel, Intel(R) HD Graphics Direct3D11 vs_5_0 ps_5_0, D3D11)"
Adapter Device Type: IntegratedGpu
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS)
Available features: Features(TEXTURE_COMPRESSION_BC | CLEAR_TEXTURE | MULTIVIEW)
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 640, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 256, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: high
Surface samples: 4
Surface size: Extent3d { width: 550, height: 400, depth_or_array_layers: 1 }

# Page Info
Page URL: chrome-extension://donbcfbmhbcapadipfkeojnmajbakjdc/player.html?url=https%3A%2F%2Fwww.shockwave.com%2Fcontent%2Fsuperspavegas%2Fsis%2FCore.swf
SWF URL: https://www.shockwave.com/content/superspavegas/sis/Core.swf

# Browser Info
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
Platform: Win32
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-07-13
Channel: nightly
Built: 2023-07-13T00:20:50.346Z
Commit: e0886707c5b9637c3fb6d3064a6ffd2cb28117de
Is extension: true

# Metadata
width: 800
height: 600
frameRate: 35
numFrames: 1
swfVersion: 8
backgroundColor: #000000
isActionScript3: false
uncompressedLength: 8006804

**Comment by Lord-McSweeney** [comment]
Should be fixed by #9433

**Comment by Lord-McSweeney** [comment]
Fixed by #9433.
